OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -32,14 +33,20 @@ public class ExplodesInfo : ITraitInfo
 		[Desc("DeathType(s) to apply upon explosion.")]
 		public readonly string[] DeathType = null;
 
-		public object Create(ActorInitializer init) { return new Explodes(this); }
+		public object Create(ActorInitializer init) { return new Explodes(init, this); }
 	}
 
 	public class Explodes : INotifyKilled
 	{
 		readonly ExplodesInfo info;
 
-		public Explodes(ExplodesInfo info) { this.info = info; }
+		public Explodes(ActorInitializer init, ExplodesInfo info)
+		{
+			this.info = info;
+
+			if (info.Weapon == null && info.EmptyWeapon == null)
```

Comment:
Could we use `[Require]` on the traitinfo instead?  I don't think it would be that big of a deal to force modders to define both the empty and loaded versions, and that gives us static guarantees.